File: src/main/java/com/kata/board/service/PostService.java
package com.kata.board.service;

import com.kata.board.entity.Post;
import com.kata.board.repository.PostRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class PostService {
    private final PostRepository postRepository;

    public List<Post> selectAll(){
        List<Post> postList = postRepository.findAll();
        return postList;
    }
}

File: src/main/kotlin/com/kata/board/service/PostService.kt
package com.kata.board.service

import com.kata.board.dto.PostDto
import com.kata.board.entity.Post

interface PostService {
    fun getPost(id: Long): PostDto

    fun getAllPost(): List<PostDto>

    fun savePost(postDto: PostDto): Post

    fun deletePost(id: Long)

    fun updatePost(postDto: PostDto): Post
}